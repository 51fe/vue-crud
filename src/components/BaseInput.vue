<template>
  <el-input
    :value="value"
    :max-length="200"
    :placeholder="placeholder"
    clearable
    class="base-input"
    @input="$emit('input', $event)"
  />
</template>

<script>
export default {
  name: 'BaseInput',
  props: {
    value: {
      type: String,
      default: ''
    },
    placeholder: {
      type: String,
      default: '请输入'
    }
  }
}
</script>

<style>
.base-input {
  width: 100%;
}
</style>
