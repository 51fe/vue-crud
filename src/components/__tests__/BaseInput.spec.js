import { fireEvent, render, screen } from '@testing-library/vue'
import BaseInput from '../BaseInput.vue'

test('renders correctly by default', () => {
  const { html } = render(BaseInput)
  screen.getByPlaceholderText('请输入')
  expect(html()).toMatchSnapshot()
})

test('changes props correctly', () => {
  const value = 'hello'
  const placeholder = '请输入姓名'
  render(BaseInput, {
    props: {
      value,
      placeholder
    }
  })
  screen.getByPlaceholderText(placeholder)
  screen.getByDisplayValue(value)
})

test('emits input event correctly', async () => {
  const { emitted } = render(BaseInput)
  const value = 'world'
  const input = screen.getByRole('textbox')
  await fireEvent.update(input, value)
  expect(emitted().input[0][0]).toBe(value)
})


