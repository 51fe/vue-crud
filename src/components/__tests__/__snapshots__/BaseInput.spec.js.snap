// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly by default 1`] = `
<div class="base-input el-input el-input--suffix">
  <!----><input type="text" autocomplete="off" max-length="200" placeholder="请输入" class="el-input__inner">
  <!---->
  <!---->
  <!---->
  <!---->
</div>
`;
